controller/userController.js

```diff
@@ -40,4 +40,52 @@ const updateUserDetails = async (req, res) => {
     res.status(500).send(err.message)
   }
 }
-module.exports = { userRegister, createFirebaseToken, updateUserDetails }
\ No newline at end of file
+const createPost = async (req, res) => {
+  const validateFieldErrors = validationResult(req);
+  if (!validateFieldErrors.isEmpty()) {
+    res.status(404).send({ validateFieldErrors })
+    return;
+  }
+  const bodyData = matchedData(req);
+  try {
+    const user = await firestore().collection("users").doc(bodyData.uid).get();
+    const createdAt = firestore.FieldValue.serverTimestamp()
+    const slugUrl = `http://localhost:8000/user/create-new-post/${slugify(bodyData.slug)}`
+    const newPost = await firestore().collection("posts").doc(bodyData.uid).set({
+      title: bodyData.title, description: bodyData.description, slug: slugUrl, updatedAt: createdAt, createdAt, updatedBy: `${user._fieldsProto.firstName.stringValue} ${user._fieldsProto.lastName.stringValue}`
+    })
+    res.status(200).send({ message: "Post created successfully", newPost })
+  } catch (err) {
+    res.status(500).send(err.message)
+  }
+}
+const uploadImage = async (req, res) => {
+  const { uid } = req.headers;
+  const file = req.file;
+  const validateFieldErrors = validationResult(req);
+  if (!validateFieldErrors.isEmpty()) {
```

Comment:
can't we move this function to a middleware? i am seeing this same function in createPost and uploadImage @ashok-choudhary-0